pyro/infer/trace_mmd.py

```diff
@@ -86,18 +86,14 @@ def kernel(self, kernel):
             # fix kernel's parameters
             for k in kernel.values():
                 if isinstance(k, pyro.contrib.gp.kernels.kernel.Kernel):
-                    for p in list(k._pyro_params.keys()):
-                        v = getattr(k, p).detach()
-                        delattr(k, p)
-                        setattr(k, p, v)
+                    for p in k.parameters():
```

Comment:
I think you can even `k.requires_grad_(False)`, but this is good already.

Reply:
Thanks for the tip!